tests/run.sh

```diff
@@ -185,4 +185,4 @@ if [[ -n $SPECIFIC_INTEGRATION_TEST ]]; then
     SPECIFIC_INTEGRATION_TEST_ARG="-k $SPECIFIC_INTEGRATION_TEST"
 fi
 
-python2 -m pytest $XDIST_ARGS $MAX_FAIL_ARG -s --verbose --junitxml=results.xml $HTML_REPORT $pass_args "$SPECIFIC_INTEGRATION_TEST_ARG" $DEFAULT_TESTS
+eval python2 -m pytest $XDIST_ARGS $MAX_FAIL_ARG -s --verbose --junitxml=results.xml $HTML_REPORT $pass_args $SPECIFIC_INTEGRATION_TEST_ARG $DEFAULT_TESTS
```

Comment:
I think this also won't work, `eval` will just make everything unquoted. Isn't the discussion about `SPECIFIC_INTEGRATION_TEST` containing spaces that must remain quoted? If so you cannot escape having two separate variables, I think, because `-k` must not be quoted together with the rest.

Reply:
It will work because `SPECIFIC_INTEGRATION_TEST` value is single quoted, both in yoctobuild-build script and in travis-ci.yml (this part in a PR, not in master).

This variable is defined like `SPECIFIC_INTEGRATION_TEST="'not Something'"` so `eval` here will do the trick (it works locally at least).

From run.sh point of view, your suggestion is the safest and cleanest. I'll rework in both repos then

Reply:
See related commit in PR: https://github.com/mendersoftware/mender-qa/pull/290/commits/140f550ffe4f030ae21b66821cd56da190f29b86